src/H5Cpkg.h

```diff
@@ -82,424 +75,167 @@
 
 #ifdef H5C_DO_SANITY_CHECKS
 
-#define H5C__DLL_PRE_REMOVE_SC(entry_ptr, head_ptr, tail_ptr, len, list_size, fail_val) \
-if ( ( (head_ptr) == NULL ) ||                                               \
-     ( (tail_ptr) == NULL ) ||                                               \
-     ( (entry_ptr) == NULL ) ||                                              \
-     ( (len) <= 0 ) ||                                                       \
-     ( (list_size) < (entry_ptr)->size ) ||                                       \
-     ( ( (entry_ptr)->prev == NULL ) && ( (head_ptr) != (entry_ptr) ) ) ||   \
-     ( ( (entry_ptr)->next == NULL ) && ( (tail_ptr) != (entry_ptr) ) ) ||   \
-     ( ( (len) == 1 ) &&                                                     \
-       ( ! ( ( (head_ptr) == (entry_ptr) ) &&                                \
-             ( (tail_ptr) == (entry_ptr) ) &&                                \
-             ( (entry_ptr)->next == NULL ) &&                                \
-             ( (entry_ptr)->prev == NULL ) &&                                \
-             ( (list_size) == (entry_ptr)->size )                                 \
-           )                                                                 \
-       )                                                                     \
-     )                                                                       \
+#define H5C__GEN_DLL_PRE_REMOVE_SC(entry_ptr, list_next, list_prev, head_ptr, tail_ptr, len, list_size, fail_val) \
+if (((head_ptr) == NULL) || ((tail_ptr) == NULL) ||                          \
+    ((entry_ptr) == NULL) || ((len) <= 0) ||                                 \
+    ((list_size) < (entry_ptr)->size) ||                                     \
+    ((entry_ptr)->list_prev == NULL && (head_ptr) != (entry_ptr)) ||        \
+    ((entry_ptr)->list_next == NULL && (tail_ptr) != (entry_ptr)) ||        \
+    ((len) == 1 &&                                                           \
+     !((head_ptr) == (entry_ptr) && (tail_ptr) == (entry_ptr) &&             \
+       (entry_ptr)->list_next == NULL && (entry_ptr)->list_prev == NULL && \
+       (list_size) == (entry_ptr)->size                                      \
+      )                                                                      \
+    )                                                                        \
    ) {                                                                       \
-    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL pre remove SC failed")     \
+    HGOTO_ERROR(H5E_CACHE, H5E_SYSTEM, (fail_val), "DLL pre insert SC failed")               \
```

Comment:
The macro says REMOVE but the error message says "insert"

Reply:
Done